src/tools/pkcs11-tool.c

```diff
@@ -5355,14 +5384,32 @@ show_key(CK_SESSION_HANDLE sess, CK_OBJECT_HANDLE obj)
 		printf("  Unique ID:  %s\n", unique_id);
 		free(unique_id);
 	}
-
+	get_token_info(opt_slot, &info);
+	printf("  uri:        %s", get_uri(&info));
+	if (id != NULL && idsize) {
+		printf(";id=%%");
+		for (unsigned int n = 0; n < idsize; n++)
+			printf("%02x", id[n]);
+		free(id);
+	}
+	if (label != NULL) {
+		const char *pelabel = percent_encode((unsigned char *)label, strlen(label));
+		printf(";object=%s", pelabel);
+		free(label);
+	}
+	if (pub) {
+		printf(";type=public\n");
```

Comment:
I hope one last nit. There is also `type=secret-key` defined by the pkcs11 uri even though its much less common and not available on yubikey, I would include it for completeness.